pymeasure/experiment/procedure.py

```diff
@@ -284,6 +284,9 @@ def emit(self, topic, record):
     def should_stop(self):
         raise NotImplementedError('should be monkey patched by a worker')
 
+    def has_next(self):
```

Comment:
I do understand your doubt regarding the name. Maybe another/clearer name would help; something like `next_measurement_queued`?

If you think a negation would be the best option (which I can understand, as often some additional work is required only if the measurement is the last one) that is also fine. You could use either with a "private" intermediate method (e.g. `_has_next`) that is called by the actual `is_last`, or maybe you could monkeypatch with a lambda or local function in the worker (e.g. `self.procedure.is_last = lambda: not self.has_next()`) if that works.

I'm fine with either choice.

Reply:
"I decided on the name ```is_last``` as I believe it's the most intuitive, although it was a difficult choice. I implemented a lambda function at the ```BaseManager``` level, so both ```Worker``` and ```Procedure``` now have is_last methods.

Reply:
Sounds good to me, nice work!